src/backend/distributed/utils/replication_origin_session_utils.c

```diff
@@ -0,0 +1,252 @@
+/*-------------------------------------------------------------------------
+ *
+ * replication_origin_session_utils.c
+ *   Functions for managing replication origin session.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "distributed/replication_origin_session_utils.h"
+#include "distributed/remote_commands.h"
+#include "distributed/metadata_cache.h"
+#include "utils/builtins.h"
+
+static bool isReplicationOriginSessionSetup(MultiConnection *connection);
+static bool isReplicationOriginCreated(MultiConnection *connection, char *originName,
+									   RepOriginId *originId);
+static RepOriginId CreateReplicationOriginSession(MultiConnection *connection,
+												  char *originName);
+static void ReplicationOriginSessionSetupHelper(MultiConnection *connection,
+												char *originName);
+static bool ExecuteRemoteCommandAndCheckResult(MultiConnection *connection, char *command,
+											   char *expected);
+
+/* SetupReplicationOriginRemoteSession sets up a new replication origin session in a
+ * remote session. The identifier is used to create a unique replication origin name
+ * for the session in the remote node.
+ */
+void
+SetupReplicationOriginRemoteSession(MultiConnection *connection, char *identifier)
+{
+	if (!isReplicationOriginSessionSetup(connection))
+	{
+		StringInfo originNameString = makeStringInfo();
+		appendStringInfo(originNameString, "citus_internal_%d_%s", GetLocalNodeId(),
+						 identifier);
+		char *originName = quote_literal_cstr(originNameString->data);
+		ReplicationOriginSessionSetupHelper(connection, originName);
+	}
+}
+
+
+/* ResetReplicationOriginRemoteSession resets the replication origin session in a
+ * remote node.
+ */
+void
+ResetReplicationOriginRemoteSession(MultiConnection *connection, char *identifier)
+{
+	/* Reset the replication origin in remote session*/
+	StringInfo replicationOriginSessionResetQuery = makeStringInfo();
+	appendStringInfo(replicationOriginSessionResetQuery,
+					 "select pg_catalog.pg_replication_origin_session_reset();");
+	ExecuteCriticalRemoteCommand(connection,
+								 replicationOriginSessionResetQuery->data);
+
+	/* Drop the replication origin entry created in remote session.*/
+	StringInfo originNameString = makeStringInfo();
+	appendStringInfo(originNameString, "citus_internal_%d_%s", GetLocalNodeId(),
+					 identifier);
```

Comment:
are there any character/length limits on replication origins?

Reply:
The "pg_replication_table" has the firsd "roname " as "text" field.
So I am not seeing any particular size/length limit in the code origin.c, replorigin_create() functions.